xdsl/dialects/builtin.py

```diff
@@ -1725,132 +1726,61 @@ def get_element_type(self) -> _UnrankedMemrefTypeElems:
 @irdl_attr_definition
 class DenseIntOrFPElementsAttr(TypedAttribute, ContainerType[AnyDenseElement]):
     name = "dense"
-    type: ParameterDef[
-        RankedStructure[IntegerType]
-        | RankedStructure[IndexType]
-        | RankedStructure[AnyFloat]
-    ]
-    data: ParameterDef[ArrayAttr[AnyIntegerAttr] | ArrayAttr[AnyFloatAttr]]
+    type: ParameterDef[RankedStructure[IntegerType | IndexType | AnyFloat]]
+    data: ParameterDef[BytesAttr]
+
+    @staticmethod
+    def get_format_str(
+        type: RankedStructure[IntegerType | IndexType | AnyFloat],
+    ) -> str:
+        data_len = prod(type.get_shape())
+        element_type = type.element_type
+
+        if isinstance(element_type, IndexType):
+            # store IndexType as long long in bytes object
+            format_str = "{}q"
```

Comment:
I think we might want to remove support for Index before this PR, as the size of index is a function of the target, not the host computer.